pallets/rmrk-core/src/functions.rs

```diff
@@ -1,5 +1,194 @@
+		ensure!(
+			NFTs::<T>::iter_prefix_values(collection_id).count() == 0,
+			Error::<T>::CollectionNotEmpty
+		);
+		Collections::<T>::remove(collection_id);
+		Ok(())
+	}
+
+	fn change_issuer(
+		collection_id: CollectionId,
+		new_issuer: T::AccountId,
+	) -> Result<(T::AccountId, CollectionId), DispatchError> {
+		ensure!(Collections::<T>::contains_key(collection_id), Error::<T>::NoAvailableCollectionId);
+
+		Collections::<T>::try_mutate_exists(collection_id, |collection| -> DispatchResult {
+			if let Some(col) = collection {
+				col.issuer = new_issuer.clone();
+			}
+			Ok(())
+		})?;
+
+		Ok((new_issuer, collection_id))
+	}
+
+	fn lock_collection(collection_id: CollectionId) -> Result<CollectionId, DispatchError> {
+		Collections::<T>::try_mutate_exists(collection_id, |collection| -> DispatchResult {
+			let collection = collection.as_mut().ok_or(Error::<T>::CollectionUnknown)?;
+			let currently_minted = NFTs::<T>::iter_prefix_values(collection_id).count();
+			collection.max = currently_minted.try_into().unwrap();
+			Ok(())
+		})?;
+		Ok(collection_id)
+	}
+}
+
+impl<T: Config> Nft<T::AccountId, StringLimitOf<T>> for Pallet<T> {
+	type CollectionId = CollectionId;
+	type NftId = NftId;
+	type MaxRecursions = T::MaxRecursions;
+
+	fn mint_nft(
+		sender: T::AccountId,
+		owner: T::AccountId,
+		collection_id: CollectionId,
+		recipient: Option<T::AccountId>,
+		royalty: Option<Permill>,
+		metadata: StringLimitOf<T>,
+	) -> sp_std::result::Result<(Self::CollectionId, Self::NftId), DispatchError> {
+		let nft_id = Self::get_next_nft_id(collection_id)?;
+
+		let collection = Self::collections(collection_id).ok_or(Error::<T>::CollectionUnknown)?;
+
+		let nfts_minted = NFTs::<T>::iter_prefix_values(collection_id).count();
+		let max: u32 = collection.max.try_into().unwrap();
+
+		ensure!(
+			nfts_minted < max.try_into().unwrap() || max == 0,
+			Error::<T>::CollectionFullOrLocked
+		);
+
+		let recipient = recipient.unwrap_or(owner.clone());
+		let royalty = royalty.unwrap_or(Permill::default());
+
+		let rootowner = owner.clone();
+		let owner_as_maybe_account = AccountIdOrCollectionNftTuple::AccountId(owner.clone());
+
+		let nft =
+			NftInfo { owner: owner_as_maybe_account, rootowner, recipient, royalty, metadata };
+
+		NFTs::<T>::insert(collection_id, nft_id, nft);
+		NftsByOwner::<T>::append(owner, (collection_id, nft_id));
+
+		Ok((collection_id, nft_id))
+	}
+
+	fn burn_nft(
+		collection_id: CollectionId,
+		nft_id: NftId,
+		max_recursions: u32,
+	) -> sp_std::result::Result<(CollectionId, NftId), DispatchError> {
+		ensure!(max_recursions > 0, Error::<T>::TooManyRecursions);
+		NFTs::<T>::remove(collection_id, nft_id);
+		if let Some(kids) = Children::<T>::take(collection_id, nft_id) {
+			for child in kids {
+				<Self as Nft<T::AccountId, StringLimitOf<T>>>::burn_nft(
+					child.0,
+					child.1,
+					max_recursions - 1,
+				)?;
+			}
```

Comment:
```suggestion
			for (child_collection_id, child_nft_id) in kids {
				<Self as Nft<T::AccountId, StringLimitOf<T>>>::burn_nft(
					child_collection_id,
					child_nft_id,
					max_recursions - 1,
				)?;
			}
```

.. for better readability. Or you can consider to accept `(CollectionId, NftId)` tuple as the function arg